umh-core/pkg/config/manager.go

```diff
@@ -142,8 +169,9 @@ func (m *FileConfigManager) GetConfigWithOverwritesOrCreateNew(ctx context.Conte
 
 // GetConfig returns the current config, always reading fresh from disk
 func (m *FileConfigManager) GetConfig(ctx context.Context, tick uint64) (FullConfig, error) {
-	m.mutexReadOrWrite.Lock()
-	defer m.mutexReadOrWrite.Unlock()
+	// we use a read lock here, because we only read the config file
```

Comment:
this will block indefnitely ignoring any cancels of the context --> need to fix that

maybe there is a GetMutexOrCancelContext function somewher